js-envs/src/main/scala/org/scalajs/jsenv/AsyncJSEnv.scala

```diff
@@ -9,11 +9,22 @@
 
 package org.scalajs.jsenv
 
-import org.scalajs.core.tools.io._
-import org.scalajs.core.tools.classpath._
-import org.scalajs.core.tools.logging._
+import org.scalajs.core.tools.io.VirtualJSFile
+import org.scalajs.core.tools.jsdep.ResolvedJSDependency
 
 trait AsyncJSEnv extends JSEnv {
-  def asyncRunner(classpath: CompleteClasspath, code: VirtualJSFile,
-      logger: Logger, console: JSConsole): AsyncJSRunner
+  def asyncRunner(libs: Seq[ResolvedJSDependency], code: VirtualJSFile): AsyncJSRunner
+
+  final def asyncRunner(code: VirtualJSFile): AsyncJSRunner =
+    asyncRunner(Nil, code)
+
+  override def loadLibs(libs: Seq[ResolvedJSDependency]): AsyncJSEnv =
+    new LoadedLibs { val loadedLibs = libs }
+
+  private[jsenv] trait LoadedLibs extends super.LoadedLibs with AsyncJSEnv {
```

Comment:
Dotty does not allow "overriding types", i.e., reintroducing a `LoadedLibs` here when there is already one in `JSEnv`. I recommend against introducing code that will be invalid in Dotty.